docs/federation/federation-directives-index.mdx

```diff
@@ -0,0 +1,31 @@
+| [`@edfs__kafkaSubscribe`](/federation/event-driven-federation/kafka)          | Declares a subscription field that listens to one or more Kafka topics. Used in EDG Subscription fields.                                           |
+| [`@edfs__natsPublish`](/federation/event-driven-federation/nats)             | Marks a mutation field that publishes an event to a NATS subject. Requires a subject and provider ID.                                              |
+| [`@edfs__natsRequest`](/federation/event-driven-federation/nats)             | Declares a query field that requests an event over NATS and expects a response. Used in EDG Query fields.                                          |
+| [`@edfs__natsSubscribe`](/federation/event-driven-federation/nats)           | Marks a subscription field that listens to one or more NATS subjects. Can include stream configuration.                                            |
+| `@external`                      | Indicates that a field (or all fields of a type) comes from another subgraph and isn’t resolved here.                                              |
+| `@extends`                       | Marks an object or interface as an extension of a type defined in another subgraph, when `extend` isn’t used.                                      |
+| `@inaccessible`                  | Hides a field or type from the API schema while keeping it in the supergraph for internal use.                                                     |
+| `@interfaceObject`               | Lets a subgraph contribute fields to an interface and all implementing types by defining an object abstraction.                                    |
+| `@key`                           | Declares one or more fields that uniquely identify an entity across subgraphs, enabling resolution.                                                |
+| `@link`                          | Connects an external spec to the schema and optionally imports types or directives for use in the subgraph.                                        |
+| [`@openfed__configureDescription`](/federation/directives/openfed__configuredescription) | Allows controlling how and whether descriptions are propagated to the federated schema. Supports overrides, hiding, and composition on extensions. |
+| [`@openfed__subscriptionFilter`](/federation/directives/openfed__subscriptionfilter)   | Applies a filter condition to a subscription field, restricting events based on field values or logical expressions.                               |
+| `@override`                      | Specifies that a field now resolves from the current subgraph instead of the one that previously defined it.                                       |
+| `@provides`                      | Indicates that specific fields can be resolved from this subgraph, but only at certain schema paths.                                               |
```

Comment:
fields specified in this directive fields argument can be resolved on a path where directive attached

fields mention in provides should be marked an external

if parent field of some nested external field are marked as external, but we have nested siblings fields which are not external, such fields will also be available for resolving on parent external paths